packages/optimizer/lib/transformers/PreloadHeroImage.js

```diff
@@ -101,9 +101,12 @@ class PreloadHeroImage {
       'as': 'image',
       'data-hero': '',
     });
-    if (heroImage.media) {
-      preload.attribs.media = heroImage.media;
+    if (!heroImage.media) {
+      // We can only safely preload a hero image if there's a media attribute
+      // as we can't detect whether it's hidden on certain viewport sizes otherwise.
```

Comment:
Why? Is it an opt in sort of thing, or does the media attr provide some kind of indication of..something?

Reply:
This is AMP's built in way to define per element media queries. The assumption is that if you use this, you won't define additional media queries in CSS. This could still lead to false positives, but I suspect that this is seldom the case.